packages/frontend/editor-ui/src/features/ai/assistant/components/Agent/AskAssistantBuild.vue

```diff
@@ -266,6 +268,41 @@ watch(
 	},
 );
 
+// Track categorization tool completion for telemetry
+watch(
+	() => builderStore.toolMessages,
+	(messages) => {
+		messages
+			.filter(
+				(msg) =>
+					msg.toolName === 'categorize_prompt' &&
+					msg.status === 'completed' &&
+					msg.toolCallId &&
+					!trackedCategorizations.value.has(msg.toolCallId),
+			)
+			.forEach((msg) => {
+				const outputUpdate = msg.updates.find((u) => u.type === 'output');
+				const categorizationData = outputUpdate?.data?.categorization as
+					| { techniques: string[]; confidence?: number }
+					| undefined;
+
+				if (categorizationData && msg.toolCallId) {
+					trackedCategorizations.value.add(msg.toolCallId);
+
+					telemetry.track('Classifier labels user prompt', {
+						user_id: usersStore.currentUserId ?? undefined,
+						workflow_id: workflowsStore.workflowId,
+						classifier_labels: categorizationData.techniques,
+						confidence: categorizationData.confidence,
+						session_id: builderStore.trackingSessionId,
+						timestamp: new Date().toISOString(),
+					});
+				}
+			});
```

Comment:
```suggestion
			.forEach((msg) => {
			const isCategorizationTool =
				msg.toolName === 'categorize_prompt' &&
				msg.status === 'completed' &&
				msg.toolCallId &&
				!trackedCategorizations.value.has(msg.toolCallId);

			if (!isCategorizationTool) return;

			const outputUpdate = msg.updates.find((u) => u.type === 'output');
			const categorizationData = outputUpdate?.data?.categorization as
				| { techniques: string[]; confidence?: number }
				| undefined;

			if (categorizationData && msg.toolCallId) {
				trackedCategorizations.value.add(msg.toolCallId);

				telemetry.track('Classifier labels user prompt', {
					user_id: usersStore.currentUserId ?? undefined,
					workflow_id: workflowsStore.workflowId,
					classifier_labels: categorizationData.techniques,
					confidence: categorizationData.confidence,
					session_id: builderStore.trackingSessionId,
					timestamp: new Date().toISOString(),
				});
			}
		});
```

This could be just a single loop